File: Spring_Boot_Web_Service/src/main/java/com/demo/service/RecipeService.java
package com.demo.service;
import java.util.List;
import java.util.Set;
import com.demo.models.Ingredient;
import com.demo.models.Recipe;
import com.demo.models.Tag;
import com.demo.models.User;

public interface RecipeService {

	List<Recipe> getAllRecipes();

	Boolean addNewRecipe(Recipe r);

	Recipe getRecipeById(int rid);

	Boolean updateRecipeById(Recipe u);

	Boolean deleteRecipeById(int rid);

	Set<Ingredient> getIngredientsByRecipe(int recipeId);

	Set<User> getUsersByRecipe(int recipeId);

	Set<Tag> getTagsByRecipe(int recipeId);

}

File: Source_Code/FF_Frontend/src/services/RecipeService.js
import axios from "axios";

const baseUrl = "http://localhost:8080/";

class RecipeService {
  getAllRecipes() {
    return axios.get(baseUrl + "recipes");
  }

  getRecipeById(recipeId) {
    return axios.get(baseUrl + "recipe/" + recipeId);
  }

  addRecipe(recipe, userId) {
    return axios.post(baseUrl + "recipe?userId=" + userId, recipe, {
      headers: {
        "Content-Type": "multipart/form-data",
      },
    });
  }

  updateRecipe(recipe) {
    return axios.put(baseUrl + "recipe/" + recipe.recipeId, recipe);
  }

  deleteRecipe(recipeId) {
    return axios.delete(baseUrl + "recipe/" + recipeId);
  }

  getUsersByRecipe(recipeId) {
    return axios.get(baseUrl + "recipe/" + recipeId + "/users");
  }

  getTagsByRecipe(recipeId) {
    return axios.get(baseUrl + "recipe/" + recipeId + "/tags");
  }

  getIngredientsByRecipe(recipeId) {
    return axios.get(baseUrl + "recipe/" + recipeId + "/ingredients");
  }
}

export default new RecipeService();
